File: app/src/prod/java/com/mycompany/myapp/app/VariantModule.java
package com.mycompany.myapp.app;

import android.support.annotation.Nullable;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLContext;
import javax.net.ssl.X509TrustManager;

import dagger.Module;
import dagger.Provides;

@Module
public class VariantModule {

    @Nullable
    @Provides
    public HostnameVerifier provideCustomHostnameVerifier() {
        // No custom implementation needed in non-dev app
        return null;
    }

    @Nullable
    @Provides
    public SSLContext provideSSLContext() {
        // No custom implementation needed in non-dev app
        return null;
    }

    @Nullable
    @Provides
    public X509TrustManager buildTrustAllTrustManager() {
        // No custom implementation needed in non-dev app
        return null;
    }
}

File: app/src/prod/java/com/mycompany/myapp/app/VariantModule.kt
package com.mycompany.myapp.app

import com.mycompany.myapp.data.OkHttpSecurityModifier
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient

@Module
class VariantModule {

    @Provides
    fun provideOkHttpClientTrustAllBinding(settings: Settings): OkHttpSecurityModifier {
        return object: OkHttpSecurityModifier {
            override fun apply(builder: OkHttpClient.Builder) {
                /* No op */
            }
        }
    }
}
